libcrux-ml-kem/c.sh

```diff
@@ -57,25 +57,30 @@ if [[ "$portable_only" = 1 ]]; then
     export LIBCRUX_DISABLE_SIMD128=1
 fi
 
+
+if [[ "$clean" = 1 ]]; then
+    cargo clean
```

Comment:
This was only cleaning C before. Let's note down somewhere that this is cleaning Rust now as well. This may lead to unexpected results because clean deletes C files that we may want to keep around.
Also, `boring.sh` cleans already itself. So that should just add clean then to this invocation.

But we can do all this in a follow up.